packages/flutter/lib/src/widgets/app.dart

```diff
@@ -462,10 +484,19 @@ class _WidgetsAppState extends State<WidgetsApp> implements WidgetsBindingObserv
       child: new Localizations(
         locale: widget.locale ?? _locale,
         delegates: _localizationsDelegates.toList(),
-        child: new Title(
-          title: widget.title,
-          color: widget.color,
-          child: result,
+        child: new Builder(
+          builder: (BuildContext context) {
+            String title = widget.title;
+            if (widget.onGenerateTitle != null) {
+              title = widget.onGenerateTitle(context);
+              assert(title != null, 'onGenerateTitle must return a non-null String');
```

Comment:
maybe if it's null default to `widget.title`?

Reply:
I think might mask an error.

Reply:
Fair enough. Document that you can't return null in the various places?

Reply:
OK, I've added a "must not return null" warning in the various places.